scalyr_agent/builtin_monitors/journald_monitor.py

```diff
@@ -412,18 +461,93 @@ def _process_entries(self):
         """
         # read all the entries
         for entry in self._journal:
-            try:
-                msg = entry.get("MESSAGE", "")
-                extra = self._get_extra_fields(entry)
+            # try:
+            msg = entry.get("MESSAGE", "")
+            extra = self._get_extra_fields(entry)
+            if not self.log_manager.emit(
+                self.format_msg("details", msg, extra), extra_fields=extra
+            ):
                 self._logger.emit_value("details", msg, extra_fields=extra)
-                self._last_cursor = entry.get("__CURSOR", None)
-            except Exception, e:
-                global_log.warn(
-                    "Error getting journal entries: %s" % str(e),
-                    limit_once_per_x_secs=60,
-                    limit_key="journald-entry-error",
+            self._last_cursor = entry.get("__CURSOR", None)
+        # except Exception, e:
+        #    global_log.warn(
+        #        "Error getting journal entries: %s" % str(e),
+        #        limit_once_per_x_secs=60,
+        #        limit_key="journald-entry-error",
+        #    )
+
+    def format_msg(
+        self, metric_name, metric_value, extra_fields=None,
+    ):
+        string_buffer = StringIO()
+        metric_name = self._force_valid_metric_or_field_name(
+            metric_name, is_metric=True
+        )
+
+        string_buffer.write(
+            "%s %s" % (metric_name, scalyr_util.json_encode(metric_value))
+        )
+
+        if extra_fields is not None:
+            for field_name in extra_fields:
+                field_value = extra_fields[field_name]
+                field_name = self._force_valid_metric_or_field_name(
```

Comment:
Actually it looks like the extra_fields is configurable, so if we want to match the existing logging behavior I think we need this.

Reply:
Right, but extra_fields is defined when the monitor starts up in config option `journal_fields`.  So, we could just do the renaming of the metrics there, instead of having to do it on every `emit`.  Also, this is code from `scalyr_logging`, right?  Let's expose that as a static method and invoke it here, rather than coding the code here.